components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/internal/AuthenticationValveServiceComponent.java

```diff
@@ -41,11 +51,33 @@
 
     @Activate
     protected void activate(ComponentContext cxt) {
+
+        loadInvalidTenantErrorPage();
         if (log.isDebugEnabled()) {
             log.debug("AuthenticationValveServiceComponent is activated");
         }
     }
 
+    private void loadInvalidTenantErrorPage() {
+
+        try {
+            Path invalidTenantDomainHtmlResponse =
+                    Paths.get(CarbonUtils.getCarbonHome(), "repository", "resources", "identity", "pages",
+                            "invalid_tenant_domain_response.html");
+            if (!Files.exists(invalidTenantDomainHtmlResponse) ||
+                    !Files.isRegularFile(invalidTenantDomainHtmlResponse)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("invalidTenantDomainHtmlResponse is not present at: " + invalidTenantDomainHtmlResponse);
+                }
+            }
+            File file = new File(invalidTenantDomainHtmlResponse.toString());
```

Comment:
Won't this throw an exception if the file is not found at the expected path?

Reply:
Fixed in 15e0d61. Added a warn log.